packages/payment-proxy-client/src/file.ts

```diff
@@ -38,3 +33,91 @@ function parseFileNameFromUrl(url: string): string {
     return "fetched-file";
   }
 }
+
+export async function fetchFileInChunks(
+  chunkSize: number,
+  baseUrl: string,
+  costPerByte: number,
+  selectedChannel: string,
+  nitroClient: NitroRpcClient
+): Promise<File> {
+  const firstChunk = await fetchFileChunk(
+    0,
+    chunkSize - 1,
+    baseUrl,
+    costPerByte,
+    selectedChannel,
+    nitroClient
+  );
+
+  const { contentLength, fileName } = firstChunk;
+  const remainingChunks = Math.ceil(contentLength / chunkSize) - 1;
+
+  const fileContents = new Uint8Array(contentLength);
+  fileContents.set(firstChunk.data);
+
+  for (let i = 1; i <= remainingChunks; i++) {
+    const start = i * chunkSize;
+    const stop = start + chunkSize - 1;
+
+    const { data } = await fetchFileChunk(
+      start,
+      stop,
+      baseUrl,
+      costPerByte,
+      selectedChannel,
+      nitroClient
+    );
+
+    fileContents.set(data, i * chunkSize);
+  }
+
+  return new File([fileContents], fileName);
+}
+
+export async function fetchFileChunk(
+  start: number,
+  stop: number,
+  baseUrl: string,
+  costPerByte: number,
+  selectedChannel: string,
```

Comment:
Nit: "selected" is a GUI concept which probably doesn't belong here. Maybe `paymentChannel` would do?

Reply:
Updated in https://github.com/statechannels/go-nitro/pull/1510/commits/25ebabf68c212c38ab76ef8a61cfd2656fb6b35d